From 9d17e66d41c5ce1d23a622c7ce9e9d6901e7cbcf Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Mon, 28 Oct 2013 17:26:08 +0800
Subject: [PATCH 26/39] ENGR00281405 Android reboot occur if do multiple
 operations at

Settings->Developer options menu. 30%

Check mOutputStream valid before call mOutputStream.close()
Check boundary before call mPointers.get(id)

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 core/java/com/android/internal/widget/PointerLocationView.java | 10 +++++++++-
 services/java/com/android/server/usb/UsbDebuggingManager.java  |  5 +++--
 2 files changed, 12 insertions(+), 3 deletions(-)

diff --git a/core/java/com/android/internal/widget/PointerLocationView.java b/core/java/com/android/internal/widget/PointerLocationView.java
index d82831f..1977a9d 100644
--- a/core/java/com/android/internal/widget/PointerLocationView.java
+++ b/core/java/com/android/internal/widget/PointerLocationView.java
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2010 The Android Open Source Project
- *
+ * Copyright (C) 2013 Freescale Semiconductor, Inc.
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
@@ -593,6 +593,14 @@ public class PointerLocationView extends View implements InputDeviceListener {
         }
         for (int i = 0; i < NI; i++) {
             final int id = event.getPointerId(i);
+
+			//add bound check
+			int size = mPointers.size();
+			if(id >= size) {
+				Log.w(TAG, "addPointerEvent, id " + id + " >= size " + size);
+				return;
+			}
+
             final PointerState ps = mCurDown ? mPointers.get(id) : null;
             final PointerCoords coords = ps != null ? ps.mCoords : mTempCoords;
             event.getPointerCoords(i, coords);
diff --git a/services/java/com/android/server/usb/UsbDebuggingManager.java b/services/java/com/android/server/usb/UsbDebuggingManager.java
index ce953a4..a3d3913 100644
--- a/services/java/com/android/server/usb/UsbDebuggingManager.java
+++ b/services/java/com/android/server/usb/UsbDebuggingManager.java
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2012 The Android Open Source Project
- *
+ * Copyright (C) 2013 Freescale Semiconductor, Inc.
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
@@ -118,7 +118,8 @@ public class UsbDebuggingManager implements Runnable {
 
     private void closeSocket() {
         try {
-            mOutputStream.close();
+			if(mOutputStream != null)
+				mOutputStream.close();
         } catch (IOException e) {
             Slog.e(TAG, "Failed closing output stream: " + e);
         }
-- 
1.8.0

